### DISCUSSION TITLE
Best practice for storing `url` and `anonKey`

### DISCUSSION THREAD

**Discussion by percarlsen** [supabase/supabase#27300] [Questions] [answered]
Hi. The [documentation](https://supabase.com/docs/reference/dart/introduction) seems to suggests hard-coding the `url` and `anonKey` variables when calling the initializer, but I'm concerned about the security of this approach. Assuming that my project has proper RLS configurations, what is the consequence if the `url` and `anonKey` variables are leaked? What is the best practice for storing these (hard-code, env variables, other approach)?

**Comment by GaryAustin1** [comment, marked as answer]
I've seen most users use env variables versus hard code just for convenience or with public repositories.

If you use browser side calls for the database/storage/functions or realtime then the url and anon key will be exposed no matter where you put them.  The jwt approach and RLS are designed to limit what anon users can do with them.   This was the default operating mode until very recently as auth calls could not easily be done server side.

Some users do not want even those out, but then you must never use them at all in browser side code or release them publicly.

**Reply by percarlsen** [reply]
Thanks for the fast answer. Sounds like it's important to be thorough with the RLS setup, then.
